### PR TITLE
cleanup pending and detail headers

### PR DESCRIPTION
**Pull request by jeking3**
Folded in pending and detail headers so they are more directly associated with dynamic_bitset.  A scan of all headers across boost shows no external usage of lowest_bit (the only header that includes it is dynamic_bitset.hpp), and detail can be moved freely.  Moved the dynamic_bitset detail out of the top level boost/detail header directory.

### CODE REVIEW

File: include/boost/dynamic_bitset/dynamic_bitset.hpp [hunk 1]
```diff
@@ -1346,6 +1346,17 @@ bool dynamic_bitset<Block, Allocator>::intersects(const dynamic_bitset & b) cons
 // --------------------------------
 // lookup
 
+template <typename T>
+int lowest_bit(T x) {
```
Comment (MarcelRaad) [inline include/boost/dynamic_bitset/dynamic_bitset.hpp]: Did you mean to duplicate lowest_bit, in its own header and here?

**Comment by jeking3** [comment]
Thanks for catching that.  I tried that first then decided just to move the
file.  I will fix that up.

**Comment by jeking3** [comment]
Apart from the coverage result oddity, this is ready for another review.  I did change the assertion to an exception to make it testable.

**Comment by swatanabe** [comment]
AMDG

On 08/22/2018 07:07 AM, James E. King III wrote:
> I did change the assertion to an exception to make it testable.
> 

  Please don't.  It's an assertion, because it's a precondition.
The behavior when violating a precondition is untestable because
it's undefined.

In Christ,
Steven Watanabe

**Comment by jeking3** [comment]
Fair enough - although assertions can be tested if the test harness can be told to expect a core.

File: include/boost/dynamic_bitset/detail/lowest_bit.hpp [hunk 2]
```diff
@@ -14,16 +14,16 @@
...
+#include "boost/assert.hpp"
 
 namespace boost {
+namespace detail {
 
     template <typename T>
     int lowest_bit(T x) {
 
-        assert(x >= 1); // PRE
+	BOOST_ASSERT(x >= 1); // PRE
```
Comment (Kojoley) [inline include/boost/dynamic_bitset/detail/lowest_bit.hpp]: You have tabs in several places

Comment (jeking3) [reply include/boost/dynamic_bitset/detail/lowest_bit.hpp]: Thanks, I will fix that up.

**Comment by pdimov** [comment]
Yes, no turning of assertions into exceptions please. :-)

If you insist on testing the BOOST_ASSERT, you can use BOOST_ENABLE_ASSERT_HANDLER, although precondition violation testing is a somewhat dubious practice.

**Pull request merged**